File: src/main/java/org/cloudfoundry/credhub/controller/v1/HealthController.java
package org.cloudfoundry.credhub.controller.v1;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import java.util.Map;

import static com.google.common.collect.ImmutableMap.of;

@Controller
public class HealthController {

  @RequestMapping(value = "/health", method = RequestMethod.GET)
  public ResponseEntity<Map> getHealth() {
    try {
      return new ResponseEntity<>(of("status", "UP"), HttpStatus.OK);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
}

File: backends/credhub/src/main/kotlin/org/cloudfoundry/credhub/health/HealthController.kt
package org.cloudfoundry.credhub.health

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@Deprecated("No longer needed after CredHub 2.2 because we have Spring Actuator")
@RestController
@RequestMapping(path = [HealthController.ENDPOINT])
class HealthController {

    companion object {
        const val ENDPOINT = "/health"
    }

    @RequestMapping(
        method = [RequestMethod.GET],
        path = [""]
    )
    fun getHealthStatus(): ResponseEntity<Map<String, String>> {
        try {
            return ResponseEntity(mapOf("status" to "UP"), HttpStatus.OK)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}
